appinventor/blocklyeditor/src/generators/yail.js

```diff
@@ -103,6 +103,13 @@ Blockly.Yail.FLONUM_REGEXP = "^[\\s]*[-+]?([0-9]*)((\\.[0-9]+)|[0-9]\\.)[\\s]*$"
  * @returns {String} the generated code if there were no errors.
  */
 Blockly.Yail.getFormYail = function(formJson, packageName, forRepl) {
+  //check for errors and warnings
+  if (Blockly.WarningHandler.getErrorCount() > 0) {
+    throw "Cannot build with errors";
```

Comment:
The string "Cannot build with errors" should be defined in a variable in _messages.js so it can be translated.